### PR TITLE
Add SystemFingerprint and chatMsg.ToolCallID field

### PR DESCRIPTION
**Pull request by gburt**
A few small improvements:
- capture the new `system_fingerprint` response field to chat calls
- on ChatCompletionMessage capture the `tool_call_id` that needs to be set for role=TOOL prompts
- fix a typo (ToolChoiche)
- Add omitempty to ChatCompletionMessage optional fields
- Add required ToolCall.type field

### CODE REVIEW

**Comment by sashabaranov** [comment]
@gburt thank you for the PR! Seems like there's a conflicting change here

**Comment by gburt** [comment]
> @gburt thank you for the PR! Seems like there's a conflicting change here

you're welcome! thanks for the great library

File: chat.go [hunk 1]
```diff
@@ -62,11 +62,17 @@ type ChatCompletionMessage struct {
 	Name string `json:"name,omitempty"`
 
 	FunctionCall *FunctionCall `json:"function_call,omitempty"`
-	ToolCalls    []ToolCall    `json:"tool_calls,omitempty"`
+
+	// For Role=ASSISTANT prompts this may be set to the tool calls generated by the model, such as function calls.
+	ToolCalls []ToolCall `json:"tool_calls,omitempty"`
```
Comment (sashabaranov) [inline chat.go:67]: Could you please point out this field in the documentation?

Comment (sashabaranov) [reply chat.go:67]: also, could we please de-capslock the `Role=ASSISTANT` and `Role=TOOL` here? :)

Comment (gburt) [reply chat.go:67]: Ah yeah good call re: caps (we use SCREAMING enums internally...).  [Here are the docs](https://platform.openai.com/docs/api-reference/chat/create) for CreateChatRequest.messages.tool_calls:
image

**Review by sashabaranov — APPROVED** [review]

**Pull request merged**